bpo-39682: make `pathlib.Path` immutable by removing (undocumented) support for "closing" a path by using it as a context manager (#18846)
python/cpython: Misc/NEWS.d/next/Library/2020-03-08-11-00-01.bpo-39682.AxXZNz.rst

> Remove undocumented support for *closing* a `pathlib.Path` object via its context manager. Support remains, but is now a no-op, making `Path` objects immutable.

Reviewer: ```suggestion
Remove undocumented support for *closing* a `pathlib.Path` object via its context manager. The context manager magic methods remain, but they are now a no-op, making `Path` objects immutable.
```

> Remove undocumented support for *closing* a `pathlib.Path` object via its context manager. The context manager magic methods remain, but they are now a no-op, making `Path` objects immutable.

Reviewer 2: Can this be split into multiple lines? The max line width for Misc/NEWS entries is [80 chars](https://devguide.python.org/committing/#what-s-new-and-news-entries):

```suggestion
Remove undocumented support for *closing* a `pathlib.Path` object via its
context manager. The context manager magic methods remain, but they are now a
no-op, making `Path` objects immutable.
```

> Removed undocumented support for *closing* a `pathlib.Path` object by using it as a context manager. Support for using a `Path` as a context manager remains, but does nothing. This change makes `Path` objects immutable.

Reviewer: ```suggestion
Remove undocumented support for *closing* a `pathlib.Path` object via its context manager. Support remains, but is now a no-op, making `Path` objects immutable.
```
